src/scheduler/resource_resv.c

```diff
@@ -304,27 +427,91 @@ dup_resource_resv_array(resource_resv **oresresv_arr,
 	server_info *nsinfo, queue_info *nqinfo)
 {
 	resource_resv **nresresv_arr;
-	int i;
+	int i, j;
+	int chunk_size;
+	th_data_dup_resresv *tdata = NULL;
+	th_task_info *task = NULL;
+	int num_tasks = 0;
+	int num_resresv;
+	int num_resresv_cpy;
+	int malloc_err = 0;
+	int tid;
 
 	if (oresresv_arr == NULL || nsinfo == NULL)
 		return NULL;
 
 	for (i = 0; oresresv_arr[i] != NULL; i++)
 		;
+	num_resresv = i;
+	num_resresv_cpy = num_resresv;
 
-	if ((nresresv_arr = malloc((i + 1) * sizeof(resource_resv *))) == NULL) {
+	if ((nresresv_arr = malloc((num_resresv + 1) * sizeof(resource_resv *))) == NULL) {
 		log_err(errno, "dup_resource_resv_array",  "Error allocating memory");
 		return NULL;
 	}
+	nresresv_arr[0] = NULL;
```

Comment:
Since you're mucking in this area of the code, change this to use \_\_func\_\_ and the macro